src/vm/object.cpp

```diff
@@ -1525,21 +1525,21 @@ void STDCALL CopyValueClassUnchecked(void* dest, void* src, MethodTable *pMT)
     switch (pMT->GetNumInstanceFieldBytes())
     {        
     case 1:
-        VolatileStore((UINT8*)dest, *(UINT8*)src);
+        *(UINT8*)dest = *(UINT8*)src;
         break;
 #ifndef ALIGN_ACCESS
         // we can hit an alignment fault if the value type has multiple 
         // smaller fields.  Example: if there are two I4 fields, the 
         // value class can be aligned to 4-byte boundaries, yet the 
         // NumInstanceFieldBytes is 8
```

Comment:
@jkotas I added a MemoryBarrier() unconditionally for now.  I will create a separate pull request with changes to make this optional